ignite/distributed/comp_models/base.py

```diff
@@ -216,10 +216,10 @@ def all_reduce(
         return cast(Union[torch.Tensor, float], self._collective_op(tensor, self._do_all_reduce, op, group=group))
 
     def all_gather(
-        self, tensor: Union[torch.Tensor, float, str], group: Optional[Any] = None
+        self, tensor: Union[torch.Tensor, float, str, Any], group: Optional[Any] = None
     ) -> Union[torch.Tensor, float, List[float], List[str]]:
         if not isinstance(tensor, (torch.Tensor, Number, str)):
-            raise TypeError(f"Unhandled input type {type(tensor)}")
+            return self._do_all_gather_object(tensor, group=group)
```

Comment:
Why don't you update and use `_collective_op`?

Reply:
`_collective_op` seemed to specific about tensor, str, Number input types. I found this code more simple to write

Reply:
What's the common thing among them(tensor, number and str) that they should have a different path? By the way, the name `_collective_op` is general.

Reply:
> What's the common thing among them(tensor, number and str) that they should have a different path?

Yes, but all_gather_object can handle tensor, number and str as well, so it is redundant. I think we can later remove special cases for number/str and use `all_*_object` ops.